### ISSUE TITLE
Zero byte at end of varchar(22) id column value with sqlite3 on Android

### ISSUE THREAD

**Issue by donv** [jruby/activerecord-jdbc-adapter#625]
Hi!

I am getting a zero byte (\u0000) at the end of a value read from sqlite3 on Android.  The column is a varchar(22) id primary key column.

Is this a known issue for sqlite3 on master?

**Comment by kares** [comment]
nope - weird ... will you be looking into this - if not I might need you to share some reproducable piece.

on a related note - I tried to look into options where I would be able to test on "normal" JDK with sqldroid's driver ... but found nothing, isn't there an emulation layer (drop-in .jar) which would allow us to run with somehow working android APIs ?

**Comment by donv** [comment]
Hi @kares !

I will be looking into the zero byte issue today.  I'll report my progres, if any :)

sqldroid should be testable with a reguler JDK.  The other team members do use it, I believe.  I myself have not used it.

**Comment by donv** [comment]
Hi @kares !

Did look at the zero byte error day before yesterday, and I _think_ we get a correct value from the database, and it is corrupted in the adapter.

I had a sick day today, but I will try looking at it again tomorrow.

I am changing jobs this week, so I cannot choose much what I do at work.

**Comment by kares** [comment]
NP, if you're not into it and someone on the team can setup a way of testing on a "normal" JDK with SQLDroid that would be very helpful (so that I do not have to figure out all of the android-like setup myself) ... I was hoping for a first 1.4 pre release - that is why I cared that much here - would be great if the pre was usable / tested with Ruboto (as well)

**Comment by donv** [comment]
Hi @kares !

I solved the problem for us!  The extra null-byte comes from the SQLite database when storing a string and reading it back as bytes or blob.  We have solved the problem in the SQLDroid JDBC driver so the correct bytes are returned.

That means that there is no change necessary in ARJDBC.  Closing.

We are testing ARJDBC master on Ruboto now, but there are still problems, mostly because of increased stack usage.  I'll file a separate issue for that.
